packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -66,6 +68,14 @@ export interface CssVarsThemeOptions extends Partial2Level<ThemeScales> {
   components?: Components<Theme>;
   colorSchemes?: Partial<Record<DefaultColorScheme | ExtendedColorScheme, ColorSystemOptions>>;
   unstable_sxConfig?: SxConfig;
+  /**
+   * A function to determine if the key, value should be attached as CSS Variable
+   * `keys` is an array that represents the object path keys.
+   *  Ex, if the theme is { foo: { bar: 'var(--test)' } }
+   *  then, keys = ['foo', 'bar']
+   *        value = 'var(--test)'
+   */
+  shouldSkipGeneratingVar?: (keys: string[], value: string | number) => boolean;
```

Comment:
This will be propagated to the theme, so that it can be used in the `CssVarsProvider`.